File: hackerrank/maximizing-xor/maximizing-xor.py
#!/usr/bin/env python

# Complete the function below.


def  maxXor( l,  r):
  r = r + 1
  maxXor = 0
  for i in xrange(l, r):
    for j in xrange(i, r):
      xorValue = i ^ j
      if (xorValue > maxXor):
        maxXor = xorValue
  return maxXor



_l = int(raw_input());


_r = int(raw_input());

res = maxXor(_l, _r);
print(res)

File: hackerrank/warmup/maximizing-xor/maximizing-xor.rb
#!/usr/bin/env ruby

l = gets.to_i
r = gets.to_i
maxXor = 0
(l..r).each {|i|
  (i+1..r).each {|j|
    xor = i ^ j
    if (xor > maxXor) then
      maxXor = xor
    end
  }
}
puts maxXor
